doc/voq/voq_hld.md

```diff
@@ -0,0 +1,809 @@
+
+# 1 Requirements Overview
+## 1.1 Functional Requirements
+
+The VOQ feature implementation shall support the following
+1.  Distributed VOQ System.
+2.  IPv4 and IPv6 unicast routing across any ports in the system
+3.  Each switch in the system is controlled by a separate "asic instance" of SONiC.
+4.  Host IP reachability to/from the interface IP addresses of any SONiC instance in the system using any network port in the system.
+5.  Host IP reachability between the SONiC instances in the system over the datapath.
+6.  Routing protocol peering between SONiC instances over the datapath.
+7.  Static provisioning of System Ports in the VOQ System Database.
+8.  Dynamic discovery of Routing Interfaces and Neighbors on other asics via the VOQ System Database.
+9.  Automatic determination of Switch_Id for each asic (optional).
+
+## 1.2 Platform requirements
+Every ASIC in the system needs be assigned a unique Switch_ID. Each ASIC consumes as many consecutive switch_id values as it has cores. So the next valid switch_id assignment should skip ahead by a number equal to the number of cores in the asic. This can be done via an additional configuration attribute. Another option is atuomatic determination of sWwitch_ID. Vendor supplied platform specific components are aware of details (line card number, asic number with in the line card, number of cores for the asic etc...) which allow them to automatically calculate the Switch_ID for each asic. For this reason it is proposed that the PMON could populate the Switch_ID for an asic into the APPL_DB.
+
+## 1.3 Configuration requirements
+Phase-1 of the Distributed VOQ System should support static configuration of the following
+1. Connection parameters for the Central VOQ System Database (IP address, Port)
+2. All the system ports in the entire system. The configuration of each system port should specify - system_port_id, switch_id, cored_index, core_port_index and speed.
+3. Maximum number of cores in the system.
+4. Switch_ID for each asic
+
+## 1.4 Orchagent requirements
+### Switch Creation
+ - Should create the switch with the correct values for max_cores and switch_id
+ - Should create the switch with complete list of system port for all the system ports during initialization
+
+### PortsOrch:
+ - Should be SystemPorts aware
+ - Should initialize system ports list and create host interfaces for local ports (including the recycle port if configured)
+
+### IntfsOrch:
+- Should create/delete/update router interfaces against local-asic ports
+- Should create/delete/update remote-asic router interfaces on the local switch
+
+### NeighOrch:
+ - Should be aware of Neighbors on other asics
+ - Should be able to create/delete/update "remote-asic" Neighbors on the local switch.
+ - Should be able to create/delete/update next hops for "remote-asic" neighbors on the local switch.
+
+### VOQ DB sync
```

Comment:
Do we still need this?

Reply:
No. Syncing is done in intfsorch and neighorch. No separate process. We'll update the hld.